[Content] Merch Computer revamp and Paintings

DeityLink opened the pull request:
![html](https://cloud.githubusercontent.com/assets/7573912/10260184/6e60a000-6972-11e5-99f6-bee32b3832b5.png)
![paintings2](https://cloud.githubusercontent.com/assets/7573912/9976083/23d57132-5ed7-11e5-9f7e-b5641a3ad308.png)
- Merch Computer sprite updated again.
- Purchases made at a Merch Computer are now spawned on top of the computer, instead of under the player.
- Merch Computers and Mining Equipment Lockers now tell your remaining balance after you've made a purchase. Just like vending machines and PDA terminals.
- The transaction history now properly features "-" in front of the amounts that were substracted from your balance.
- Tweaked the pricing of the Merch Computer's products. Overall everything has been made much cheaper.
- Revamped the Merch Computer's UI. It is now an HTML interface similar to the PDA Terminal's.
- Added categories to the Merch Computer, and a few items to fill them. Including an item that previously couldn't be produced, the Photo Album, as a Luxury item.
- The most expensive items to purchase on the Merch Computer are now paintings. You get a random painting everytime you purchase one.
- Paintings can be freely placed and picked up from walls.
- People dabbling or expert in pixel art are welcome to contribute their paintings (just ping me on IRC, quote me in the thread, or ahelp when I'm on the server).
- Abandonned Crates on the asteroid now have a chance to contain a painting.
- Added a random painting to the safes of every map's Vault.
- Added a random painting to the Captain's Office on every map.
- Added a Justice painting to the Courtroom on every map.
- Mr.Deempisi's portrait in the Bar has be converted into a painting. You can now pick it up and place it elsewhere without needing a screwdriver.
- Nar-Sie turns every paintings into PaPa NarNar
- Those also have a chance to spawn in Hidden Temples on the asteroid.
- Nar-Sie now replaces reagent dispensers (fuel tanks, water tanks, beer kegs) with old kegs, filled with blood.
- Cultists can finally examine talismans to check which rune they've been imbued with. You no longer have any excuse for attempting to stun that sec officer with the supply talisman!

PJB3005 commented on code/__HELPERS/unsorted.dm:
```diff
@@ -1366,6 +1366,13 @@ proc/find_holder_of_type(var/atom/reference,var/typepath) //Returns the first ob
 /proc/print_runtime(exception/e)
 	world.log << "[time_stamp()] Runtime detected\n[e] at [e.file]:[e.line]\n [e.desc]"
 
+/proc/transfer_fingerprints(atom/A,atom/B)
```
I'm 100% sure this already exists somewhere.

PJB3005 commented on code/__HELPERS/unsorted.dm:
```diff
@@ -1366,6 +1366,13 @@ proc/find_holder_of_type(var/atom/reference,var/typepath) //Returns the first ob
+/proc/transfer_fingerprints(atom/A,atom/B)
+	if(!A || !B)
+		return
+	B.fingerprints = A.fingerprints
```
You need to do `A.fingerprints.Copy()`.

DeityLink replied:
Out of curiosity how is Copy() better?

PJB3005 replied:
@DeityLink since these variables are lists, you're not copying the list, you're literally setting B's fingerprints var equal to A's, if you were to modify them on either B or A, the other atom would get the modified prints too.

DeityLink replied:
Thanks for the explanation. Though in the case of the paintings, it works out since they are meant to be the same item (both the structure on the wall and the frame that you pick up).
If there is already another proc that copies fingerprints, then we can keep this one to "link" the fingerprints of two atoms.

PJB3005 replied:
Could use some clarification just in case though.

PJB3005 commented on code/__HELPERS/unsorted.dm:
```diff
@@ -1366,6 +1366,13 @@ proc/find_holder_of_type(var/atom/reference,var/typepath) //Returns the first ob
+	if(!A || !B)
+		return
+	B.fingerprints = A.fingerprints
+	B.fingerprintshidden = A.fingerprintshidden
```
Same.

PJB3005 commented on code/__HELPERS/unsorted.dm:
```diff
@@ -1366,6 +1366,13 @@ proc/find_holder_of_type(var/atom/reference,var/typepath) //Returns the first ob
+		return
+	B.fingerprints = A.fingerprints
+	B.fingerprintshidden = A.fingerprintshidden
+	B.fingerprintslast = A.fingerprintslast
```
Whoosh

Probe1 commented:
> It is now a NanoUI interface

gross, we have enough issues with lag.  i'm not saying don't merge because of it but we need to move off nano and to html or something different.

clusterfack commented:
Shouldn't cause a particularly significant issue with nano, it only has 10 elements, though yes it would be far better if you made this HTML instead Deity

DeityLink commented:
@Probe1 @clusterfack give me a simple template to copy to have it work in HTML and I'll do it. I'm a decipherer remember.

PJB3005 commented:
@DeityLink the TeG is a good example IMO.

DeityLink commented:
Removed the nanoUI stuff, we're now using html interfaces....or at least trying to.

clicking on the computer opens a script error window, clicking either yes or no will make the items appear, and they can be purchased normally, but the window has no formatting.

![help](https://cloud.githubusercontent.com/assets/7573912/10217881/126e78ac-6833-11e5-83a5-b8ffc620b7b9.png)

I'll need you guys to help me there.

Also added two more pictures. One is a reference to Undertale, the other is an original creation by me.

![paintings](https://cloud.githubusercontent.com/assets/7573912/10217926/562834e8-6833-11e5-811e-bd232254c823.png)

PJB3005 commented on code/game/machinery/computer/store.dm:
```diff
@@ -3,9 +3,72 @@
+/obj/machinery/computer/merch/Destroy()
+	..()
+	html_machines -= src
+
```
You need to `qdel()` the interface datum here or it won't GC correctly.

PJB3005 commented on code/game/machinery/computer/store.dm:
```diff
@@ -20,115 +83,62 @@
+		return
+
+	var/delay = 0
+	delay += send_asset(user.client, "Chart.js")
```
What, where are you using a chart in this UI?

DeityLink replied:
I have no idea what I am doing. I just write code and pray that it works.

PJB3005 replied:
Just remove references to `Chart.JS`, it's the javascript library used for the power monitor's chart.

PJB3005 commented on code/game/machinery/computer/store.dm:
```diff
@@ -162,3 +169,22 @@ td.cost.toomuch {
 			icon_state = initial(icon_state)
 			stat &= ~NOPOWER
+
+/obj/machinery/computer/merch/process()
```
WHAT THE FUCK.

WHY.

PJB3005 commented on code/game/machinery/computer/store.dm:
```diff
@@ -20,115 +83,62 @@
+	spawn(delay) //To prevent Jscript issues with resource sending.
+		interface.show(user)
 
+/*
```
Since you're uncommenting it just kill it entirely

PJB3005 commented on code/game/machinery/computer/store.dm:
```diff
@@ -3,9 +3,72 @@
+				overflow: hidden;
+				width: auto;
+			}
+		<script src="Chart.js"></script>
```
Why?

PJB3005 commented on code/game/machinery/computer/store.dm:
```diff
@@ -3,9 +3,72 @@
+	..()
+
+	var/head = {"
+		<style type="text/css">
```
Found your Jscript error, you're missing this 'style' tag's closing tag.

PJB3005 commented on code/game/objects/items/mountable_frames/paintings.dm:
```diff
@@ -0,0 +1,205 @@
+			desc = "a blank painting."
+
+/obj/structure/painting/attack_hand(mob/user)
+	user << "<span class='notice'>You pick up the [src]...</span>"
```
Should be `\the`

DeityLink commented:
ok we're good

PJB3005 commented on code/modules/power/monitor.dm:
```diff
@@ -81,24 +81,25 @@
 /obj/machinery/power/monitor/Destroy()
 	..()
 	html_machines -= src
+	qdel(interface)
```
er?

DeityLink replied:
Isn't that what you wanted?

PJB3005 replied:
But it's on the power monitor...

and it's qdel()'d 2 lines later
